src/addr.rs

```diff
@@ -198,6 +198,16 @@ impl VirtAddr {
     pub const fn p4_index(self) -> PageTableIndex {
         PageTableIndex::new_truncate((self.0 >> 12 >> 9 >> 9 >> 9) as u16)
     }
+
+    /// Returns the 9-bit level page table index.
+    ///
+    /// ## Panics
+    /// Panics if level is not between 1 and 4
+    #[inline]
+    pub fn p_index(self, level: u8) -> PageTableIndex {
+        assert!(1 <= level && level <= 4, "level has to be between 1 and 4",);
```

Comment:
This function can be make `const fn`.

You will need to do some array indexing weirdness to get the assert! to work, but something like the following should work:
```rust
#[cfg(feature = "const_fn")]
assert!(1 <= level && level <= 4, "level has to be between 1 and 4");
#[cfg(not(feature = "const_fn"))] 
[(); 1][!(1 <= level && level <= 4) as usize];
```

We do something similar here:
https://github.com/rust-osdev/x86_64/blob/8060f05c1314467520052a8d70f9de8ef80e4aa0/src/structures/gdt.rs#L120-L126

Note: this will also fix the Clippy linit.

Reply:
That's a neat trick. Thanks for the suggestion :+1: